api/server/httputils/errors.go

```diff
@@ -72,6 +73,12 @@ func GetHTTPErrorStatusCode(err error) int {
 				break
 			}
 		}
+
+		// if the error has a type of grpc.rpcError, the error message should
+		// contain a sub string of "rpc error: code = "
+		if strings.Contains(errStr, "rpc error: code = ") {
```

Comment:
Basically, move this above this stuff:
```
			{"not found", http.StatusNotFound},
			{"cannot find", http.StatusNotFound},
			{"no such", http.StatusNotFound},
			{"bad parameter", http.StatusBadRequest},
			{"no command", http.StatusBadRequest},
			{"conflict", http.StatusConflict},
			{"impossible", http.StatusNotAcceptable},
			{"wrong login/password", http.StatusUnauthorized},
			{"unauthorized", http.StatusUnauthorized},
			{"hasn't been activated", http.StatusForbidden},
			{"this node", http.StatusServiceUnavailable},
			{"needs to be unlocked", http.StatusServiceUnavailable},
			{"certificates have expired", http.StatusServiceUnavailable},
```

And only check that stuff if the parsed statusCode from the grpc check is 500.